vss_collection.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 Robert Bosch GmbH and Microsoft Corporation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from typing import List
+
+from vspec.model.vsstree import VSSNode
+
+from utils import CodeGeneratorContext
+
+_COLLECTION_SUFFIX = "Collection"
+_COLLECTION_REG_EX = r"\w+\[\d+,(\d+)\]"
+
+_TYPE_SUFFIX = "Type"
+_DEFAULT_RANGE_NAME = "element"
+
+"""VSS Collection helper."""
+
+
+class VssInstance:
+    """VSS Instance Model."""
+
+    def __init__(self, name: str, content: List[str], is_range: bool):
+        """Construct of new insntance object."""
+        # Content List
+        self.content = content
+
+        # The name of the Instance
+        self.name = name
+
+        # Is a named range flag that will be used to generate
+        # getter function for element by index.
+        self.is_range = is_range
+
+
+class VssCollection:
+    """VSS Collection Object."""
+
+    def __init__(self, node: VSSNode, ctx: CodeGeneratorContext):
```

Comment:
do you really need this context here? you could also create a new one.

Reply:
created a new one.